### ISSUE TITLE
toProto3Json with option to serialize enums as ints

### ISSUE THREAD

**Issue by ignatz** [google/protobuf.dart#932]
Hi folks,

In the wild i've encountered a bunch of proto implementations that don't seem to agree on whether proto->json should render enum values as strings or ints. This one uses strings.

I'm not sure if there is a standard or whether it's under specified. I guess this all goes back to machines talk binary, humans read text :shrug: This mantra has served me well so far for machine to machine comms, however now I wanted to use Postgres' json support and I'm starting to notice this entire world i was previously blissfully ignorant towards.

Long story short, I was wondering if it was possible for your or I to add support for enum to integer serialization? I checked, it's a 5 letter change s/name/value/: https://github.com/google/protobuf.dart/blob/ccf104dbc36929c0f8708285d5f3a8fae206343e/protobuf/lib/src/protobuf/proto3_json.dart#L42 (plus some wiring to make it switchable and some golden tests)

EDIT: I forgot to mention that the deserialization side already handles the int case: https://github.com/google/protobuf.dart/blob/ccf104dbc36929c0f8708285d5f3a8fae206343e/protobuf/lib/src/protobuf/proto3_json.dart#L215 , maybe this makes my request sound my question less outrageous :) 

Best

**Comment by sigurdm** [comment]
https://protobuf.dev/programming-guides/proto3/#json specifies that serialization should always use strings, and deserialization should accept both.

That being said, I think that having it being configurable is reasonable.

**Comment by ignatz** [comment]
That was the piece I was missing... eye opening. I had stumbled already on some of the other particularities of the spec as well, such as int64 -> String. I guess, JavaScript only has ~53bit integer precision (ignoring BigIng given JSON) but still leads to interesting challenges when dealing with various number types.

Anyway, I should probably revisit my approach of using proto as a json schema. Consider this issue closed and thanks for the quick reply. Much appreciated.

Closing thought and spec aside, if the implementation would "leak" a few of its guts a bit more liberally one could implement their own serializer extension method :woman_shrugging:
